Microsoft.Toolkit.HighPerformance/Memory/Span2D{T}.cs

```diff
@@ -0,0 +1,669 @@
+        }
+
+        /// <inheritdoc cref="Span{T}.GetHashCode()"/>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        [Obsolete("GetHashCode() on Span will always throw an exception.")]
+        public override int GetHashCode()
+        {
+            throw new NotSupportedException("Microsoft.Toolkit.HighPerformance.Span2D<T>.GetHashCode() is not supported");
+        }
+#pragma warning restore CS0809
+
+        /// <summary>
+        /// Fills the elements of this span with a specified value.
+        /// </summary>
+        /// <param name="value">The value to assign to each element of the <see cref="Span2D{T}"/> instance.</param>
+        public void Fill(T value)
+        {
+            if (this.pitch == 0)
+            {
+                MemoryMarshal.CreateSpan(ref this.reference.Value, Length).Fill(value);
+            }
+            else
+            {
+                ref T sourceRef = ref this.reference.Value;
+                IntPtr step = (IntPtr)(this.width + this.pitch);
+
+                // Fill each row individually
+                for (int i = 0; i < this.height; i++)
+                {
+                    MemoryMarshal.CreateSpan(ref sourceRef, this.width).Fill(value);
+
+                    sourceRef = ref Unsafe.Add(ref sourceRef, step);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns an enumerator for the current <see cref="Span2D{T}"/> instance.
+        /// </summary>
+        /// <returns>
+        /// An enumerator that can be used to traverse the items in the current <see cref="Span2D{T}"/> instance
+        /// </returns>
+        [Pure]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Enumerator GetEnumerator() => new Enumerator(this);
+
+        /// <summary>
+        /// Returns a reference to the 0th element of the <see cref="Span2D{T}"/> instance. If the current
+        /// instance is empty, returns a <see langword="null"/> reference. It can be used for pinning
+        /// and is required to support the use of span within a fixed statement.
+        /// </summary>
+        /// <returns>A reference to the 0th element, or a <see langword="null"/> reference.</returns>
+        [Pure]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public unsafe ref T GetPinnableReference()
+        {
+            ref T r0 = ref Unsafe.AsRef<T>(null);
+
+            if (this.Length != 0)
+            {
+                r0 = ref this.reference.Value;
+            }
+
+            return ref r0;
+        }
+
+        /// <summary>
+        /// Returns a reference to the first element within the current instance, with no bounds check.
+        /// </summary>
+        /// <returns>A reference to the first element within the current instance.</returns>
+        [Pure]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public ref T DangerousGetReference()
```

Comment:
Probably shouldn't be directly on the type given its dangerous nature

Reply:
That's fine in this package, it's basically all full of dangerous APIs anyway. `MemoryOwner<T>` and `SpanOwner<T>` for instance already expose this same API too directly on the type.